xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-ui/src/main/resources/js/security/requiredrights/requiredRightsDialog.js

```diff
@@ -0,0 +1,579 @@
+        {
+            // TODO: verify if this is all correct and required.
+            this.dialogElement = document.createElement('div');
+            this.dialogElement.className = 'modal fade';
+            this.dialogElement.id = 'required-rights-dialog';
+            this.dialogElement.tabIndex = -1;
+            this.dialogElement.role = 'dialog';
+            this.dialogElement.setAttribute('aria-labelledby', 'required-rights-dialog-label');
+            this.dialogElement.setAttribute('aria-hidden', 'true');
+            this.dialogElement.innerHTML = `
+                        <div class="modal-dialog" role="document">
+                            <div class="modal-content">
+                                <div class="modal-header">
+                                <!-- TODO: check if we don't have a standard for the close button now. -->
+                                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                                        <span aria-hidden="true">&times;</span>
+                                    </button>
+                                    <h2 class="modal-title" id="required-rights-dialog-label"></h2>
+                                </div>
+                                <div class="modal-body">
+                                    <div class="enforce-selection"></div>
+                                    <div class="rights-selection">
+                                        <h3></h3>
+                                        <p></p>
+                                        <ul></ul>
+                                    </div>
+                                    <div class="required-rights-advanced-toggle-container">
+                                        <a href="#" class="required-rights-advanced-toggle"
+                                            aria-controls="required-rights-results" aria-expanded="false">
+                                        <!-- TODO: use icon theme. -->
+                                            <span class="icon-collapsed"><i class="fa fa-caret-right"></i></span>
+                                            <span class="icon-expanded"><i class="fa fa-caret-down"></i></span>
+                                        </a>
+                                    </div>
+                                    <div id="required-rights-results" class="hidden" 
+                                        aria-labelledby="advanced-toggle" aria-expanded="false">
+                                    </div>
+                                </div>
+                                <div class="modal-footer">
+                                    <button type="button" class="btn btn-secondary" data-dismiss="modal"></button>
+                                    <button type="button" class="btn btn-primary"></button>
+                                </div>
+                            </div>
+                        </div>
+                    `;
```

Comment:
hmmm wouldn't it be cleaner to have that html in a template loaded on demand?

Reply:
I've the feeling that we need to define a best practice for this kind of need. Maybe using Vue in this kind of situation would be possible? I've no idea.